### DISCUSSION TITLE
Bandmap - Suggestion for a visual supplement

### DISCUSSION THREAD

**Discussion by dl2ki** [foldynl/QLog#334] [Ideas]
The bandmap displays the band range that is specified under "Settings, Bands". 

In this area, the Spots are displayed according to the DX cluster filter setting. It often happens that stations are displayed in the bandmap that are outside the selected mode section (here CW).

One idea would be to leave the display as it is, another would be to mark the band sections visually to make orientation easier. You could mark all band sections in the display area of the bandmap, or just the section for the selected mode, which I personally think is the better solution.

In my opinion, the display should be unobtrusive so as not to visually overload the GUI of the bandmap and also so as not to increase the dimensions of the bandmap window. I have created some examples of this, but I don't particularly like all of them.

![Screenshot_20240229_143019_a](https://github.com/foldynl/QLog/assets/26311402/0566128f-4934-451e-96aa-ff22ca1053d6)
![Screenshot_20240229_143019_b](https://github.com/foldynl/QLog/assets/26311402/b1e7dea9-3b1d-4c85-a278-efcf8497246f)
![Screenshot_20240229_143019_c](https://github.com/foldynl/QLog/assets/26311402/3eb0ce5c-cd98-400e-8dc6-853460cb9a56)

In my view, these examples seem too overloaded. It is also sufficient to only select the section for the selected mode.

I would like the small gray / or colored stripes and especially the last example.

![Screenshot_20240229_143019_d1](https://github.com/foldynl/QLog/assets/26311402/c10b0b22-2901-42ba-a0e1-56325a4afcc3)
![Screenshot_20240229_143019_d2](https://github.com/foldynl/QLog/assets/26311402/756719f0-31e0-4809-9ee7-41c2864bb131)
![Screenshot_20240229_143019_d2a](https://github.com/foldynl/QLog/assets/26311402/4b48696d-f3d0-4cf7-929d-afd67f193643)
![Screenshot_20240229_143019_d2b](https://github.com/foldynl/QLog/assets/26311402/4092a1c1-9ae3-4dc3-91c8-512e25b5e858)

But perhaps there are other, better-looking variants.

It is just an idea for an inconspicuous addition to the bandmap and is not necessary for the function. As with everything, it's a matter of opinion. So you don't have to implement it.

**Comment by foldynl** [comment]
it looks good at first glance. However, I have to wonder if it can be implemented with regard to the IARU regions and their bandplans. QLog is starting to be used around the world and I'm just waiting for someone to write that the QLog's internal bandplan is incorrect because he/she is from a different region than Europe. Briefly I have seen, there are no big differences in the bandplans, but I have to look at it in more detail. 

Thanks for the your idea.

**Comment by dl2ki** [comment]
Yes, that is generally a problem. It's also something that many people don't need. It only makes sense if I actually work via the QLog bandmap.

From my point of view, you can also do without an automatic system here and use a manual frequency table, similar to CQRLog, for the few standard modes in the shortwave band plan. Everyone can then enter the specific ranges they need in the table.

Otherwise you would generally have to store all band plans in the database, which in my view would be too time-consuming in relation to the benefits. 

I have no overview of what operating modes are required for the higher frequency ranges. I have also not looked to see whether and how this is implemented in the bandmap in other logbook programs.

It would be visually clear if only the area that is actually selected as a mode was marked discreetly.

But this is not an urgent feature, so you have any time to think about it.

**Comment by rsaue** [comment]
Hi, 
I am a bit late to this conversation, but there is a way to to this you have not discussed, which is to make the mode-coloring of the bandmap a function of a (user adjusted) band plan. This is how it is done in a few other logging program I know of, and In LogHX3 which is my mostly used logger when operating in Windows. 

I enclose a screenshot of the bandmap widget and of the band plan set up screen, where the user, among other values, can define the color shown in the band map.

73 de LB2EG Richard 

![bandmap](https://github.com/foldynl/QLog/assets/22942440/00978b01-fb69-4598-b1db-3cef7a574bce)

![band setup](https://github.com/foldynl/QLog/assets/22942440/9968680f-f00f-4201-a6fa-c8bfd67d123c)

**Comment by foldynl** [comment]
I like your first image (bandmap). Not so much the second image, because that's exactly what I wouldn't want QLog to be - the settings of each parameter of application. The bad thing is that without the second image I am not able to accurately implement the beautiful bandmap in the first image - see the reasons above. Another question is what about the segments that are "shared" - multiple mods are allowed. 

This discussion is definitely worth noting

**Comment by foldynl** [comment]
I have tested both coloring variants, and I think the variant below is more noticeable. What do you think? If the color was all over the bandmap, it affected the readability of the callsigns.

![image](https://github.com/foldynl/QLog/assets/942769/95760d1d-af33-4e04-ba00-90e72a107fbc) 
![image](https://github.com/foldynl/QLog/assets/942769/ca6a1a74-b7c8-49a4-9f02-221eb02bc935)

**Comment by dl2ki** [comment]
From my point of view, the design of the bandmap should fit into the overall GUI. I like the clear design of the QLog interface, without the usual overload of colors in the Windows world. These usually make the interfaces rather confusing and restless.

The GUI design of QLog is therefore more oriented towards the habits of Linux users and therefore differs pleasantly from some Windows programs.

The complete coloring of the bandmap, as shown in Figure 1, goes in this direction. If this is implemented, it should be possible to switch it off individually in the settings.

The narrow strip, as in the last two pictures, is very clear and easier to recognize. It is also not necessary to color all the strip segments, as you only need the segment that you have selected as the mode.  This reduces the coloring to just one color, as in the last two pictures. It also significantly reduces the effort required to set individual band segments depending on the region.

All in all, I am a fan of clear and uncluttered user interfaces with as little color design as possible. Hence this individual opinion of mine. 

I think the wishes of many Windows users go in a different direction. If you also want to develop the GUI in this direction, you should make it possible to select the previous GUI design in the settings.
